drivers/backup/portworx/portworx.go

```diff
@@ -1535,6 +1617,166 @@ func (p *portworx) DeleteBackupSchedulePolicy(orgID string, policyList []string)
 	return nil
 }
 
+// CreateRuleForBackupWithMultipleApplications creates backup rule for multiple application in one rule
+func (p *portworx) CreateRuleForBackupWithMultipleApplications(orgID string, appList []string, ctx context.Context, appParameters ...map[string]backup.AppRule) (string, string, error) {
+	var (
+		preUid             string
+		preRuleName        string
+		postRuleName       string
+		postUid            string
+		preActionValue     []string
+		preContainer       []string
+		postActionValue    []string
+		postContainer      []string
+		postBackground     []bool
+		postRunInSinglePod []bool
+		preBackground      []bool
+		preRunInSinglePod  []bool
+		preRulesInfo       api.RulesInfo
+		postRulesInfo      api.RulesInfo
+		prePodSelector     []map[string]string
+		postPodSelector    []map[string]string
+	)
+	if len(appParameters) == 0 {
+		appParameters = append(appParameters, AppRuleMaster)
+	}
+
+	for i := 0; i < len(appList); i++ {
+		if reflect.DeepEqual(appParameters[0][appList[i]].PreRule, backup.PreRule{}) {
+			log.Infof("Pre rule not required for application %v", appList[i])
+		} else {
+			for j := 0; j < len(appParameters[0][appList[i]].PreRule.Rule.PodSelectorList); j++ {
+				ps := strings.Split(appParameters[0][appList[i]].PreRule.Rule.PodSelectorList[j], "=")
+				psMap := make(map[string]string)
+				psMap[ps[0]] = ps[1]
+				prePodSelector = append(prePodSelector, psMap)
+				preActionValue = append(preActionValue, appParameters[0][appList[i]].PreRule.Rule.ActionList[j])
+				backgroundVal, _ := strconv.ParseBool(appParameters[0][appList[i]].PreRule.Rule.Background[j])
+				preBackground = append(preBackground, backgroundVal)
+				podVal, _ := strconv.ParseBool(appParameters[0][appList[i]].PreRule.Rule.RunInSinglePod[j])
+				preRunInSinglePod = append(preRunInSinglePod, podVal)
+				containerName := fmt.Sprintf("%s-%s", "container", appList[i])
+				preContainer = append(preContainer, os.Getenv(containerName))
+			}
+		}
+
+		if reflect.DeepEqual(appParameters[0][appList[i]].PostRule, backup.PostRule{}) {
+			log.Infof("Post rule not required for application %v", appList[i])
```

Comment:
For some reason the `reflect.DeepEqual` doesn't seem to work. But we can create a separate common function which checks if a field exists in the interface.

I have tried it here - https://go.dev/play/p/otE34z6d6m3

Please check

Reply:
But I see the log getting printed  for postgres which does not have post rule
17:56:54  2023-07-25 12:26:54 +0000:[INFO] [{BasicBackupCreation}] [portworx.(*portworx).CreateRuleForBackupWithMultipleApplications:#1671] - Post rule not required for application postgres-backup

Reply:
reflect.DeepEqual(appParameters[0][appList[i]].PostRule, backup.PostRule{}) verifies if appParameters[0][appList[i]] DOES NOT have PostRule field